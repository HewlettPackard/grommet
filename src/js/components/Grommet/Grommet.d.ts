import * as React from 'react';
import Grommet from '../index';

export declare namespace GrommetTypes {
  type Direction = 'rtl' | 'ltr';
}

export interface GrommetProps extends Grommet.Props {
  dir: GrommetTypes.Direction;
  theme: {};
}

export class Grommet extends React.Component<GrommetProps, undefined> { }

export default Grommet;
