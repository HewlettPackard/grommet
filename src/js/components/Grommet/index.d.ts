import * as React from "react";

export interface GrommetProps {
  full?: boolean;
  theme?: object;
}

declare const Grommet: React.StatelessComponent<GrommetProps>;

export { Grommet };
