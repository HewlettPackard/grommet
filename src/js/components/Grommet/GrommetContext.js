import React from 'react';

// Track which components mounted onto the DOM are coming from
// Grommet (see FocusedContainer.js)
export const GrommetContext = React.createContext(undefined);
