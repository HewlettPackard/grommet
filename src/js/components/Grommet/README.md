## Grommet
This is the top level Grommet container.

[![](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/grommet/grommet-site?initialpath=grommet&amp;module=%2Fscreens%2FGrommet.js)
## Usage

```javascript
import { Grommet } from 'grommet';
<Grommet>...</Grommet>
```

## Properties

**dir**

Whether text should be rendered right to left or not. Defaults to
      inherit from the document context.

```
rtl
ltr
```

**theme**

Custom styles for Grommet app component.

```
object
```
  