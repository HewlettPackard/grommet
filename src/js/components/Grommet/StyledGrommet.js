import styled, { css } from 'styled-components';

import { baseStyle } from '../../utils';

const fullStyle = css`
  width: 100vw;
  height: 100vh;
`;

const StyledGrommet = styled.div`
  ${baseStyle}
  ${props => props.full && fullStyle}
  ${props => props.theme.global.font.face}
`;

export default StyledGrommet.extend`
  ${props => props.theme.grommet.extend}
`;
