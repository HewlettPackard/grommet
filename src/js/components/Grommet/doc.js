import { describe, PropTypes } from 'react-desc';

import { getAvailableAtBadge } from '../../utils';

export default (Grommet) => {
  const DocumentedGrommet = describe(Grommet)
    .availableAt(getAvailableAtBadge('Grommet'))
    .description('This is the top level Grommet container.')
    .usage(
      `import { Grommet } from 'grommet';
<Grommet>...</Grommet>`
    );

  DocumentedGrommet.propTypes = {
    dir: PropTypes.oneOf(['rtl', 'ltr']).description(
      `Whether text should be rendered right to left or not. Defaults to
      inherit from the document context.`
    ),
    theme: PropTypes.object.description('Custom styles for Grommet app component.'),
  };

  return DocumentedGrommet;
};
