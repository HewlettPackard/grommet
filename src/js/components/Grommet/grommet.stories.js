import React from 'react';
import { storiesOf } from '@storybook/react';

import { Grommet, Anchor, Box } from 'grommet';
import { Add } from 'grommet-icons';

const customTheme = {
  global: {
    colors: {
      custom: '#cc6633',
    },
  },
};


const Themed = () => (
  <Grommet theme={customTheme}>
    <Box pad='medium'>
      <Anchor icon={<Add />} label='Add' color='custom' />
    </Box>
  </Grommet>
);

storiesOf('Grommet', module)
  .add('Theme', () => <Themed />);
