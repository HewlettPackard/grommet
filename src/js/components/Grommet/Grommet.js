import React, { forwardRef, useEffect, useMemo, useState } from 'react';
import { createGlobalStyle } from 'styled-components';

import {
  ContainerTargetContext,
  ResponsiveContext,
  ThemeContext,
} from '../../contexts';

import {
  deepMerge,
  backgroundIsDark,
  getBreakpoint,
  getDeviceBreakpoint,
  normalizeColor,
} from '../../utils';
import { base as baseTheme } from '../../themes';
import { StyledGrommet } from './StyledGrommet';

const FullGlobalStyle = createGlobalStyle`
  body { margin: 0; }
`;

const deviceResponsive = (userAgent, theme) => {
  // log('--deviceResponsive', userAgent, theme);
  /*
   * Regexes provided for mobile and tablet detection are meant to replace
   * a full-featured specific library due to contributing a considerable size
   * into the bundle.
   *
   * User agents found https://deviceatlas.com/blog/list-of-user-agent-strings
   */
  if (userAgent) {
    if (/(tablet|ipad|playbook|silk)|(android(?!.*mobile))/i.test(userAgent)) {
      return getDeviceBreakpoint('tablet', theme);
    }
    if (/Mobile|iPhone|Android/.test(userAgent)) {
      return getDeviceBreakpoint('phone', theme);
    }
    return getDeviceBreakpoint('computer', theme);
  }
  return undefined;
};

const Grommet = forwardRef((props, ref) => {
  const {
    children,
    full,
    containerTarget = typeof document === 'object' ? document.body : undefined,
    theme: themeProp,
    ...rest
  } = props;

  const { background, dir, themeMode, userAgent } = props;

  const [stateResponsive, setResponsive] = useState();

  const theme = useMemo(() => {
    const nextTheme = deepMerge(baseTheme, themeProp || {});

    const {
      colors: { background: themeBackground },
    } = nextTheme.global;

    nextTheme.dark = (themeMode || nextTheme.defaultMode) === 'dark';
    const color = normalizeColor(background || themeBackground, nextTheme);
    nextTheme.dark = backgroundIsDark(color, nextTheme);
    nextTheme.baseBackground = background || themeBackground;
    // This allows DataTable to intelligently set the background of a pinned
    // header or footer.
    nextTheme.background = nextTheme.baseBackground;

    if (dir) {
      nextTheme.dir = dir;
    }

    return nextTheme;
  }, [background, dir, themeMode, themeProp]);

  useEffect(() => {
    const onResize = () => {
      setResponsive(getBreakpoint(window.innerWidth, theme));
    };
    window.addEventListener('resize', onResize);
    onResize();
    return () => {
      window.removeEventListener('resize', onResize);
    };
  }, [theme]);

  const responsive =
    stateResponsive ||
    deviceResponsive(userAgent, theme) ||
    theme.global.deviceBreakpoints.tablet;

  return (
    <ThemeContext.Provider value={theme}>
      <ResponsiveContext.Provider value={responsive}>
        <ContainerTargetContext.Provider value={containerTarget}>
          <StyledGrommet full={full} {...rest} ref={ref}>
            {children}
          </StyledGrommet>
          {full && <FullGlobalStyle />}
        </ContainerTargetContext.Provider>
      </ResponsiveContext.Provider>
    </ThemeContext.Provider>
  );
});

Grommet.displayName = 'Grommet';

let GrommetDoc;
if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  GrommetDoc = require('./doc').doc(Grommet);
}
const GrommetWrapper = GrommetDoc || Grommet;

export { GrommetWrapper as Grommet };
