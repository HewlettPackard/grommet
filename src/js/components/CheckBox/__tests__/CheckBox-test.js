import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Grommet } from '../../Grommet';
import { CheckBox } from '../';

test('CheckBox renders', () => {
  const component = renderer.create(
    <Grommet>
      <CheckBox />
      <CheckBox id='test id' name='test name' />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CheckBox label renders', () => {
  const component = renderer.create(
    <Grommet>
      <CheckBox label='test label' />
      <CheckBox label={<div>test label</div>} />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CheckBox checked renders', () => {
  const component = renderer.create(
    <Grommet>
      <CheckBox checked={true} />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CheckBox disabled renders', () => {
  const component = renderer.create(
    <Grommet>
      <CheckBox disabled={true} />
      <CheckBox disabled={true} checked={true} />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CheckBox reverse renders', () => {
  const component = renderer.create(
    <Grommet>
      <CheckBox reverse={true} label='test label' />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CheckBox toggle renders', () => {
  const component = renderer.create(
    <Grommet>
      <CheckBox toggle={true} />
      <CheckBox toggle={true} checked={true} />
      <CheckBox toggle={true} label='test label' />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
