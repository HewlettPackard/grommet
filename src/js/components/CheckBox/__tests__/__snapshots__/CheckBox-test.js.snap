// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox checked renders 1`] = `
.c0 {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 > div:first-child {
  margin-right: 12px;
}

.c1:hover input:not([disabled]) + div,
.c1:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c2 {
  opacity: 0;
  width: 0;
  height: 0;
  margin: 0;
}

.c2:focus + div,
.c2:focus + span {
  border-color: #00CCEB;
  box-shadow: 0 0 2px 2px #00CCEB;
}

.c2:focus + div > circle,
.c2:focus + span > circle,
.c2:focus + div > ellipse,
.c2:focus + span > ellipse,
.c2:focus + div > line,
.c2:focus + span > line,
.c2:focus + div > path,
.c2:focus + span > path,
.c2:focus + div > polygon,
.c2:focus + span > polygon,
.c2:focus + div > polyline,
.c2:focus + span > polyline,
.c2:focus + div > rect,
.c2:focus + span > rect {
  outline: #00CCEB solid 2px;
}

.c2:checked + div {
  border-color: #865CD6;
}

.c2:checked + div > svg {
  display: block;
}

.c2:checked + span > span {
  left: 24px;
  background-color: #865CD6;
}

.c3 {
  position: relative;
  top: -1px;
  display: inline-block;
  width: 24px;
  height: 24px;
  vertical-align: middle;
  background-color: inherit;
  border: 2px solid;
  border-color: rgba(0,0,0,0.15);
  border-radius: 4px;
}

.c3 > svg {
  position: absolute;
  top: -2px;
  left: -2px;
  display: none;
  width: 24px;
  height: 24px;
  stroke-width: 4px;
  stroke: #865CD6;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor={undefined}
  >
    <div
      className=""
    >
      <input
        checked={true}
        className="c2"
        disabled={undefined}
        id={undefined}
        name={undefined}
        onChange={undefined}
        tabIndex="0"
        type="checkbox"
      />
      <div
        className="c3"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,11.3 L10.3,16 L18,6.2"
            fill="none"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`CheckBox disabled renders 1`] = `
.c0 {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 > div:first-child {
  margin-right: 12px;
}

.c1:hover input:not([disabled]) + div,
.c1:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c2 {
  opacity: 0;
  width: 0;
  height: 0;
  margin: 0;
}

.c2:focus + div,
.c2:focus + span {
  border-color: #00CCEB;
  box-shadow: 0 0 2px 2px #00CCEB;
}

.c2:focus + div > circle,
.c2:focus + span > circle,
.c2:focus + div > ellipse,
.c2:focus + span > ellipse,
.c2:focus + div > line,
.c2:focus + span > line,
.c2:focus + div > path,
.c2:focus + span > path,
.c2:focus + div > polygon,
.c2:focus + span > polygon,
.c2:focus + div > polyline,
.c2:focus + span > polyline,
.c2:focus + div > rect,
.c2:focus + span > rect {
  outline: #00CCEB solid 2px;
}

.c2:checked + div {
  border-color: #865CD6;
}

.c2:checked + div > svg {
  display: block;
}

.c2:checked + span > span {
  left: 24px;
  background-color: #865CD6;
}

.c3 {
  position: relative;
  top: -1px;
  display: inline-block;
  width: 24px;
  height: 24px;
  vertical-align: middle;
  background-color: inherit;
  border: 2px solid;
  border-color: rgba(0,0,0,0.15);
  border-radius: 4px;
}

.c3 > svg {
  position: absolute;
  top: -2px;
  left: -2px;
  display: none;
  width: 24px;
  height: 24px;
  stroke-width: 4px;
  stroke: #865CD6;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor={undefined}
  >
    <div
      className=""
    >
      <input
        checked={undefined}
        className="c2"
        disabled={true}
        id={undefined}
        name={undefined}
        onChange={undefined}
        tabIndex="0"
        type="checkbox"
      />
      <div
        className="c3"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,11.3 L10.3,16 L18,6.2"
            fill="none"
          />
        </svg>
      </div>
    </div>
  </label>
  <label
    className="c1"
    htmlFor={undefined}
  >
    <div
      className=""
    >
      <input
        checked={true}
        className="c2"
        disabled={true}
        id={undefined}
        name={undefined}
        onChange={undefined}
        tabIndex="0"
        type="checkbox"
      />
      <div
        className="c3"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,11.3 L10.3,16 L18,6.2"
            fill="none"
          />
        </svg>
      </div>
    </div>
    <input
      name={undefined}
      type="hidden"
      value="true"
    />
  </label>
</div>
`;

exports[`CheckBox label renders 1`] = `
.c0 {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 > div:first-child {
  margin-right: 12px;
}

.c1:hover input:not([disabled]) + div,
.c1:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c2 {
  opacity: 0;
  width: 0;
  height: 0;
  margin: 0;
}

.c2:focus + div,
.c2:focus + span {
  border-color: #00CCEB;
  box-shadow: 0 0 2px 2px #00CCEB;
}

.c2:focus + div > circle,
.c2:focus + span > circle,
.c2:focus + div > ellipse,
.c2:focus + span > ellipse,
.c2:focus + div > line,
.c2:focus + span > line,
.c2:focus + div > path,
.c2:focus + span > path,
.c2:focus + div > polygon,
.c2:focus + span > polygon,
.c2:focus + div > polyline,
.c2:focus + span > polyline,
.c2:focus + div > rect,
.c2:focus + span > rect {
  outline: #00CCEB solid 2px;
}

.c2:checked + div {
  border-color: #865CD6;
}

.c2:checked + div > svg {
  display: block;
}

.c2:checked + span > span {
  left: 24px;
  background-color: #865CD6;
}

.c3 {
  position: relative;
  top: -1px;
  display: inline-block;
  width: 24px;
  height: 24px;
  vertical-align: middle;
  background-color: inherit;
  border: 2px solid;
  border-color: rgba(0,0,0,0.15);
  border-radius: 4px;
}

.c3 > svg {
  position: absolute;
  top: -2px;
  left: -2px;
  display: none;
  width: 24px;
  height: 24px;
  stroke-width: 4px;
  stroke: #865CD6;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor={undefined}
  >
    <div
      className=""
    >
      <input
        checked={undefined}
        className="c2"
        disabled={undefined}
        id={undefined}
        name={undefined}
        onChange={undefined}
        tabIndex="0"
        type="checkbox"
      />
      <div
        className="c3"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,11.3 L10.3,16 L18,6.2"
            fill="none"
          />
        </svg>
      </div>
    </div>
    <div>
      test label
    </div>
  </label>
  <label
    className="c1"
    htmlFor={undefined}
  >
    <div
      className=""
    >
      <input
        checked={undefined}
        className="c2"
        disabled={undefined}
        id={undefined}
        name={undefined}
        onChange={undefined}
        tabIndex="0"
        type="checkbox"
      />
      <div
        className="c3"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,11.3 L10.3,16 L18,6.2"
            fill="none"
          />
        </svg>
      </div>
    </div>
    <div>
      test label
    </div>
  </label>
</div>
`;

exports[`CheckBox renders 1`] = `
.c0 {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 > div:first-child {
  margin-right: 12px;
}

.c1:hover input:not([disabled]) + div,
.c1:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c2 {
  opacity: 0;
  width: 0;
  height: 0;
  margin: 0;
}

.c2:focus + div,
.c2:focus + span {
  border-color: #00CCEB;
  box-shadow: 0 0 2px 2px #00CCEB;
}

.c2:focus + div > circle,
.c2:focus + span > circle,
.c2:focus + div > ellipse,
.c2:focus + span > ellipse,
.c2:focus + div > line,
.c2:focus + span > line,
.c2:focus + div > path,
.c2:focus + span > path,
.c2:focus + div > polygon,
.c2:focus + span > polygon,
.c2:focus + div > polyline,
.c2:focus + span > polyline,
.c2:focus + div > rect,
.c2:focus + span > rect {
  outline: #00CCEB solid 2px;
}

.c2:checked + div {
  border-color: #865CD6;
}

.c2:checked + div > svg {
  display: block;
}

.c2:checked + span > span {
  left: 24px;
  background-color: #865CD6;
}

.c3 {
  position: relative;
  top: -1px;
  display: inline-block;
  width: 24px;
  height: 24px;
  vertical-align: middle;
  background-color: inherit;
  border: 2px solid;
  border-color: rgba(0,0,0,0.15);
  border-radius: 4px;
}

.c3 > svg {
  position: absolute;
  top: -2px;
  left: -2px;
  display: none;
  width: 24px;
  height: 24px;
  stroke-width: 4px;
  stroke: #865CD6;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor={undefined}
  >
    <div
      className=""
    >
      <input
        checked={undefined}
        className="c2"
        disabled={undefined}
        id={undefined}
        name={undefined}
        onChange={undefined}
        tabIndex="0"
        type="checkbox"
      />
      <div
        className="c3"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,11.3 L10.3,16 L18,6.2"
            fill="none"
          />
        </svg>
      </div>
    </div>
  </label>
  <label
    className="c1"
    htmlFor="test id"
  >
    <div
      className=""
    >
      <input
        checked={undefined}
        className="c2"
        disabled={undefined}
        id="test id"
        name="test name"
        onChange={undefined}
        tabIndex="0"
        type="checkbox"
      />
      <div
        className="c3"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,11.3 L10.3,16 L18,6.2"
            fill="none"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`CheckBox reverse renders 1`] = `
.c0 {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 > div:first-child {
  margin-left: 12px;
}

.c1:hover input:not([disabled]) + div,
.c1:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c2 {
  opacity: 0;
  width: 0;
  height: 0;
  margin: 0;
}

.c2:focus + div,
.c2:focus + span {
  border-color: #00CCEB;
  box-shadow: 0 0 2px 2px #00CCEB;
}

.c2:focus + div > circle,
.c2:focus + span > circle,
.c2:focus + div > ellipse,
.c2:focus + span > ellipse,
.c2:focus + div > line,
.c2:focus + span > line,
.c2:focus + div > path,
.c2:focus + span > path,
.c2:focus + div > polygon,
.c2:focus + span > polygon,
.c2:focus + div > polyline,
.c2:focus + span > polyline,
.c2:focus + div > rect,
.c2:focus + span > rect {
  outline: #00CCEB solid 2px;
}

.c2:checked + div {
  border-color: #865CD6;
}

.c2:checked + div > svg {
  display: block;
}

.c2:checked + span > span {
  left: 24px;
  background-color: #865CD6;
}

.c3 {
  position: relative;
  top: -1px;
  display: inline-block;
  width: 24px;
  height: 24px;
  vertical-align: middle;
  background-color: inherit;
  border: 2px solid;
  border-color: rgba(0,0,0,0.15);
  border-radius: 4px;
}

.c3 > svg {
  position: absolute;
  top: -2px;
  left: -2px;
  display: none;
  width: 24px;
  height: 24px;
  stroke-width: 4px;
  stroke: #865CD6;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor={undefined}
  >
    <div
      className=""
    >
      <input
        checked={undefined}
        className="c2"
        disabled={undefined}
        id={undefined}
        name={undefined}
        onChange={undefined}
        tabIndex="0"
        type="checkbox"
      />
      <div
        className="c3"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,11.3 L10.3,16 L18,6.2"
            fill="none"
          />
        </svg>
      </div>
    </div>
    <div>
      test label
    </div>
  </label>
</div>
`;

exports[`CheckBox toggle renders 1`] = `
.c0 {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 > div:first-child {
  margin-right: 12px;
}

.c1:hover input:not([disabled]) + div,
.c1:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c2 {
  opacity: 0;
  width: 0;
  height: 0;
  margin: 0;
}

.c2:focus + div,
.c2:focus + span {
  border-color: #00CCEB;
  box-shadow: 0 0 2px 2px #00CCEB;
}

.c2:focus + div > circle,
.c2:focus + span > circle,
.c2:focus + div > ellipse,
.c2:focus + span > ellipse,
.c2:focus + div > line,
.c2:focus + span > line,
.c2:focus + div > path,
.c2:focus + span > path,
.c2:focus + div > polygon,
.c2:focus + span > polygon,
.c2:focus + div > polyline,
.c2:focus + span > polyline,
.c2:focus + div > rect,
.c2:focus + span > rect {
  outline: #00CCEB solid 2px;
}

.c2:checked + div {
  border-color: #865CD6;
}

.c2:checked + div > svg {
  display: block;
}

.c2:checked + span > span {
  left: 24px;
  background-color: #865CD6;
}

.c3 {
  position: relative;
  vertical-align: middle;
  display: inline-block;
  width: 48px;
  height: 24px;
  border: 2px solid;
  border-color: rgba(0,0,0,0.15);
  border-radius: 24px;
}

.c4 {
  position: absolute;
  top: -2px;
  left: -2px;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  width: 24px;
  height: 24px;
  background-color: #d9d9d9;
  border-radius: 24px;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor={undefined}
  >
    <div
      className=""
    >
      <input
        checked={undefined}
        className="c2"
        disabled={undefined}
        id={undefined}
        name={undefined}
        onChange={undefined}
        tabIndex="0"
        type="checkbox"
      />
      <span
        className="c3"
      >
        <span
          className="c4"
        />
      </span>
    </div>
  </label>
  <label
    className="c1"
    htmlFor={undefined}
  >
    <div
      className=""
    >
      <input
        checked={true}
        className="c2"
        disabled={undefined}
        id={undefined}
        name={undefined}
        onChange={undefined}
        tabIndex="0"
        type="checkbox"
      />
      <span
        className="c3"
      >
        <span
          className="c4"
        />
      </span>
    </div>
  </label>
  <label
    className="c1"
    htmlFor={undefined}
  >
    <div
      className=""
    >
      <input
        checked={undefined}
        className="c2"
        disabled={undefined}
        id={undefined}
        name={undefined}
        onChange={undefined}
        tabIndex="0"
        type="checkbox"
      />
      <span
        className="c3"
      >
        <span
          className="c4"
        />
      </span>
    </div>
    <div>
      test label
    </div>
  </label>
</div>
`;
